### PR TITLE
Groundedness improvments and dotnet examples

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
### Motivation and Context
PR Feedback for https://github.com/microsoft/semantic-kernel/pull/1064

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/skills/GroundingSkill/ReferenceCheckEntities/skprompt.txt [hunk 1]
```diff
@@ -55,13 +55,18 @@ And a realio, trulio, little pet dragon
...
 {{$input}}
+</context>
 </input>
 
 <grounding_context>
 {{$reference_context}}
 </grounding_context>
 
-Please put your response after the "Response:". Produce a bulleted list of the <ungrounded_entities> which are not grounded in the context, and then repeat the <context>
+Please put your response after the "Response:". Produce a bulleted list of the <ungrounded_entities> which are not grounded in the context.
```
Comment (riedgar-ms) [inline samples/skills/GroundingSkill/ReferenceCheckEntities/skprompt.txt:63]: Given this change, does the example need updating above?

Comment (riedgar-ms) [reply samples/skills/GroundingSkill/ReferenceCheckEntities/skprompt.txt:63]: (Sorry, GitHub won't let me highlight the lines directly.... L46-51

File: samples/skills/GroundingSkill/ReferenceCheckEntities/config.json [hunk 2]
```diff
@@ -13,13 +13,18 @@
...
+        "description": "The text to check against a grounding reference.",
         "defaultValue": ""
       },
       {
         "name": "reference_context",
         "description": "The reference context to be used to ground the entities. Only those missing from the reference_context will be returned",
         "defaultValue": ""
+      },
+      {
+        "name": "entities",
```
Comment (riedgar-ms) [inline samples/skills/GroundingSkill/ReferenceCheckEntities/config.json:25]: How does this get into the context and made available?

File: samples/dotnet/kernel-syntax-examples/Example48_GroundednessChecks.cs [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+            "GroundingSkill");
+
+        kernel.ImportSkill(new TextSkill());
+
+        var config = new SequentialPlannerConfig { };
+        var planner = new SequentialPlanner(kernel, config);
+        var plan = await planner.CreatePlanAsync(ask);
+        Console.WriteLine(plan.ToPlanString());
+
+        var results = await plan.InvokeAsync(s_groundingText);
```
Comment (riedgar-ms) [inline samples/dotnet/kernel-syntax-examples/Example48_GroundednessChecks.cs:147]: This seems to be rather different from how plans in Python work? The plan there has no 'invoke' method. And the planner itself just takes a plan and a kernel. I don't see how to pass in the grounding text as an argument to the plan like you're doing here.

**Review by riedgar-ms — COMMENTED** [review]
Thanks, but I'm having trouble figuring out how to bring this over to Python

File: samples/skills/GroundingSkill/ReferenceCheckEntities/skprompt.txt (hunk 1 above)
Comment (lemillermicrosoft) [reply samples/skills/GroundingSkill/ReferenceCheckEntities/skprompt.txt:63]: Good catch!

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**